### DISCUSSION TITLE
Navigating Map Tile Zoom Transitions and Image Display

### DISCUSSION THREAD

**Discussion by jocieldo** [mapsforge/vtm#1053] [Q&A] [answered]
What about zoom levels in the context of ITileDataSource? Let's consider an example where I have an mbtiles map with zoom levels ranging from 15 to 20. Specifically, let's focus on the tiles at zoom level 17. When we move from zoom level 17 to 18, is the tile that was originally at level 17 simply being magnified to fit level 18 (like 17.1, 17.2, 17.3,...,18)? In other words, does the image between zoom levels 17 and 18 consist of the image from level 17 that has been zoomed in?

Is there a way to avoid magnifying the tile from level 17 when transitioning to level 18? Instead, can we retrieve the images from zoom level 18, but with a zoom-out effect, so that they fit within the level 18 tile grid without directly magnifying the level 17 image?

There is any place I can look into it?
Thank you!

**Comment by devemux86** [comment, marked as answer]
Map engines work like this: display existing data, until new data is available.

You could check the "bitmap" classes, for example [here](https://github.com/mapsforge/vtm/tree/master/vtm/src/org/oscim/tiling/source/bitmap) or [here](https://github.com/mapsforge/vtm/tree/master/vtm/src/org/oscim/layers/tile/bitmap).
Could extend or write new classes for a custom implementation.

**Reply by jocieldo** [reply]
Perfect, I'm gonna take a look on that. Thanks!
